bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+        id = connectable.to_terse()
+
+        dialects = cast(Iterable[Dialect] | None, ctx.params.get("dialects"))
+
+        try:
+            with tarfile.open(fileobj=input) as tar:
+                members = tar.getmembers()
+
+                if not any(
+                    member.name.startswith(f"./{id}/") for member in members
+                ):
+                    STDERR.print(
+                        f"Couldn't find a '{id}' directory in {input.name}.",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                versions: frozenset[str] = frozenset()
+                try:
+                    versions_content = tar.extractfile(
+                        f"./{id}/matrix-versions.json",
+                    )
+                    if versions_content:
+                        versions = json.load(versions_content)
+                except KeyError:
+                    STDERR.print(
+                        "No versions detected (couldn't find a "
+                        "'matrix-versions.json' file in ",
+                        f"{id} directory of {input.name}.)",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                versions_dirs = [
+                    member.name
+                    for member in members
+                    if member.isdir()
+                    and (
+                        member.name.removeprefix(f"./{id}/").removeprefix("v")
+                    )
+                    in versions
+                ]
+                if not versions_dirs:
+                    STDERR.print(
+                        f"Couldn't find any versions sub-directories in {id} "
+                        f"directory of {input.name} as per listed by its "
+                        "matrix-versions.json file.",
+                    )
+                    ctx.exit(EX.DATAERR)
+
+                progress = Progress(
+                    TextColumn("[bold blue]{task.description}"),
+                    SpinnerColumn(finished_text=""),
+                    BarColumn(bar_width=None),
+                    TextColumn(
+                        "[progress.percentage]{task.percentage:>3.0f}%",
+                    ),
+                    "•",
+                    MofNCompleteColumn(),
+                    "•",
+                    TimeElapsedColumn(),
+                    console=console.Console(),
+                    transient=True,
+                )
+                dialects = dialects or Dialect.known()
+                pretty_names_str = pretty_names_str_for(dialects)
+                total_files = len(versions) * len(list(dialects))
+                task = progress.add_task(
+                    description=(
+                        f"Preparing to parse all versioned reports of "
+                        f"{id} for {pretty_names_str} found in {input.name}"
+                    ),
+                    total=total_files,
+                )
+
+                actual_parsed_files = [0]
```

Comment:
Why is this a list, you don't appear to ever append to it or anything other than dealing with the one int.